src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -13,22 +13,46 @@
 static OpResult NodeByLabelScanInit(OpBase *opBase);
 static Record NodeByLabelScanConsume(OpBase *opBase);
 static Record NodeByLabelScanConsumeFromChild(OpBase *opBase);
+static Record NodeByLabelScanNoOp(OpBase *opBase);
 static OpResult NodeByLabelScanReset(OpBase *opBase);
 static void NodeByLabelScanFree(OpBase *opBase);
 
 static inline int NodeByLabelScanToString(const OpBase *ctx, char *buf, uint buf_len) {
 	return ScanToString(ctx, buf, buf_len, ((const NodeByLabelScan *)ctx)->n);
 }
 
+// Checks to see if operation index is within its bounds.
+static inline bool _outOfBounds(NodeByLabelScan *op) {
+	/* Because currentId starts at minimum and only increases
+	 * we only care about top bound. */
+	if(op->currentId > op->maxId) return true;
+	if(op->currentId == op->maxId && !op->maxInclusive) return true;
+	return false;
+}
+
+static inline void _setIndices(NodeByLabelScan *op) {
+	// The largest possible entity ID is the same as Graph_RequiredMatrixDim.
+	op->maxId = MIN(op->maxId, Graph_RequiredMatrixDim(op->g));
+	op->currentId = op->minId;
+	/* Advance current ID when min is not inclusive and */
```

Comment:
Change to:
`/* Advance current ID when min is not inclusive. */`
Provided that we don't need to worry about logic around `maxId == ID_RANGE_UNBOUND`? Is that a safe assumption or no?

Reply:
added protection
thanks